charts/kong/templates/_helpers.tpl

```diff
@@ -1600,3 +1600,13 @@ networking.k8s.io/v1beta1
 extensions/v1beta1
 {{- end -}}
 {{- end -}}
+
+{{- define "kong.proxy.compatibleReadiness" -}}
+{{- $proxyReadiness := .Values.readinessProbe -}}
+{{- if (or (semverCompare "< 3.3.0" (include "kong.effectiveVersion" .Values.image)) (semverCompare "< 2.11.0" (include "kong.effectiveVersion" .Values.ingressController.image))) -}}
```

Comment:
I think we should check KIC's version only if `.Values.ingressController.enabled` to make it work when the controller is not used, shouldn't we?
```suggestion
{{- if (or (semverCompare "< 3.3.0" (include "kong.effectiveVersion" .Values.image)) (and .Values.ingressController.enabled (semverCompare "< 2.11.0" (include "kong.effectiveVersion" .Values.ingressController.image)))) -}}
```